### PR TITLE
chore: consolidate and simplify isort config files

### PR DESCRIPTION
**Pull request by dzhu**
## Description

The exact duplicates of the config file seem unnecessary, and we can make use of isort's profile feature to avoid specifying several of the other options.

The imports of DeepSpeed move because isort now (correctly) sees it as a third-party package (the determination appears to be based on the directory containing the config file). Similarly, some other packages no longer need to be specified.

## Test Plan

- [x] `make fmt`

### CODE REVIEW

File: examples/.isort.cfg [hunk 1]
```diff
@@ -1,12 +0,0 @@
-[settings]
-skip_glob = **/__init__.py
-default_section = THIRDPARTY
-ensure_newline_before_comments = true
-skip = tutorials/core_api_pytorch_mnist/model_def_distributed.py
```
Comment (rb-determined-ai) [inline examples/.isort.cfg:5]: wait, we just added this... it looks like it's being removed with this change?

Comment (rb-determined-ai) [reply examples/.isort.cfg:5]: or... that's probably what the `profile=black` bit does I suppose

**Comment by rb-determined-ai** [comment]
any chance you could let the [big deepspeed PR](https://github.com/determined-ai/determined/pull/6924/files#) land before this?  I suspect they will conflict.

**Comment by tayritenour** [comment]
> any chance you could let the [big deepspeed PR](https://github.com/determined-ai/determined/pull/6924/files#) land before this? I suspect they will conflict.

It looks like it might be fixing an issue we are seeing with the PR: https://github.com/determined-ai/determined/actions/runs/5092973369/jobs/9155014037?pr=6924

Some weird desync with what isort wants on the git hooks vs normal isort

File: examples/.isort.cfg (hunk 1 above)
Comment (dzhu) [reply examples/.isort.cfg:5]: Yes, the `profile = black` includes setting `ensure_newline_before_comments`, and the `skip` doesn't appear to be necessary as long as one of those is present.

**Comment by dzhu** [comment]
I can wait if you really want, especially if that's on track to come in pretty soon, but then we'll need to put up another fix for the current GHA failures, and it will be a pretty straightforward conflict to resolve anyway.

**Comment by rb-determined-ai** [comment]
I'll let @tayritenour make that call.  I've approved the pr in question, fwiw

**Review by hamidzr — APPROVED** [review]
lgtm assuming the automated checks are happy. thanks

**Review by tayritenour — APPROVED** [review]
yeah go ahead and merge this, I'll deal with any fallout on the `feat-deepspeed-autotune` branch

**Pull request merged**